Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -138,7 +139,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               M_xface(i, j, k, 2) += dt * (-PhysConst::mu0 * mag_gamma_interp) * ( M_xface(i, j, k, 0) * Hy_eff - M_xface(i, j, k, 1) * Hx_eff)
                 + dt * Gil_damp * ( M_xface(i, j, k, 0) * ( M_xface(i, j, k, 2) * Hx_eff - M_xface(i, j, k, 0) * Hz_eff)
                 - M_xface(i, j, k, 1) * ( M_xface(i, j, k, 1) * Hz_eff - M_xface(i, j, k, 2) * Hy_eff));
- 
+
+              // temporary normalized magnitude of M_xface field at the fixed point
+              // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
+              Real M_normalized = sqrt( pow(M_xface(i, j, k, 0),2.0) + pow(M_xface(i, j, k, 1),2.0) + pow(M_xface(i, j, k, 2),2.0) ) /
+                                  MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+
+#ifdef AMREX_DEBUG
+              // check the normalized error
+              if ( std::abs(1.-M_normalized) > M_normalized_error ){
+                  printf("i = %d, j=%d, k=%d\n", i, j, k);
+                  amrex::Abort("Exceed the normalized error of the M_xface field");
+              }
+#endif
+              // normalize the M_xface field
+              M_xface(i,j,k,0) = M_xface(i,j,k,0)/M_normalized;
+              M_xface(i,j,k,1) = M_xface(i,j,k,1)/M_normalized;
+              M_xface(i,j,k,2) = M_xface(i,j,k,2)/M_normalized;
```

Comment:
change these to "/=" to give the compiler optimization hints?  (same for M_yface and M_zface)

Reply:
alternatively, if you define, amrex::Real inv_mag_normalized = 1.0_rt/M_normalized, then you can do
M_xface(i,j,k,0) *= inv_mag_normalized

Reply:
Thanks for the suggestions. I will use /= here.